schema/crdb/dbinit.sql

```diff
@@ -3660,7 +3672,15 @@ CREATE TABLE IF NOT EXISTS omicron.public.vmm (
     sled_id UUID NOT NULL,
     propolis_ip INET NOT NULL,
     propolis_port INT4 NOT NULL CHECK (propolis_port BETWEEN 0 AND 65535) DEFAULT 12400,
-    state omicron.public.vmm_state NOT NULL
+    state omicron.public.vmm_state NOT NULL,
+    failure_reason omicron.public.vmm_failure_reason,
```

Comment:
Note that a `NULL` here (or, in Rust, a `None`) can mean either "VMM is not `Failed`" *or* "no specific failure reason". Which, admittedly, feels a little bit weird. I considered alternatively adding an "other" variant to `VmmFailureReason`, and requiring that any `Failed` instance or VMM *must* have a failure reason...but I'm not sure if that actually gets us anything of value. We're still using the `Failed` *state* to conclusively indicate that an instance/VMM is `Failed`.

I suppose the main benefit is that for an instance's `last_failure_reason`, if we differentiate between `NULL` and "other", then the presence of an "other" can indicate that the instance has failed in the past, while `NULL` could mean "has never failed". Hmm...

Reply:
Okay, I think I've convinced myself that we should have an `Other` variant for the reason of, if we add this to the external API `Instance` type, UIs could use the last failure reason field to show why an instance previously failed even if it has been restarted. If "any other reason" is `NULL` and "hasn't failed" is also `NULL`, they can't easily do that. I'll go change it.